### PR TITLE
Implement Pagination Backend

### LINKED ISSUE #56: Implement reusable pagination component
Like the approach for [a shared component for all items](https://github.com/OWASP/Nest/issues/54), we want a pagination component that works across all pages (Contribute, Projects, Chapters, and Committees).

This pagination component should be flexible and configurable, allowing it to connect with the backend and provide the right pagination features for each page.

### PR DESCRIPTION
**Pull request by harsh3dev**
Resolves #56 

* This pull request introduces pagination to all API endpoints in the `backend/apps/owasp/api/search` 
* add page parameter
* include `nbPages` for getting total number of pages

### CODE REVIEW

**Review by Rajgupta36 — CHANGES_REQUESTED** [review]
hi @harsh3dev ,please set args page=1 by default and use the page-1 logic

File: backend/apps/owasp/api/search/chapter.py [hunk 1]
```diff
@@ -8,7 +8,7 @@
 from apps.owasp.models.chapter import Chapter
 
 
-def get_chapters(query, attributes=None, limit=25, meta=None):
+def get_chapters(query, attributes=None, limit=25, meta=None, page=0):
```
Comment (arkid15r) [inline backend/apps/owasp/api/search/chapter.py]:
```suggestion
def get_chapters(query, attributes=None, limit=25, meta=None, page=1):
```

File: backend/apps/owasp/api/search/chapter.py [hunk 2]
```diff
@@ -26,23 +26,25 @@ def get_chapters(query, attributes=None, limit=25, meta=None):
         "hitsPerPage": limit,
         "minProximity": 4,
         "typoTolerance": "min",
+        "page": page,
```
Comment (arkid15r) [inline backend/apps/owasp/api/search/chapter.py]:
```suggestion
        "page": page - 1,
```

**Review by arkid15r — CHANGES_REQUESTED** [review]
I agree `page=1` by default is more readable.

Could you apply the changes everywhere in the code?

Other than that looks good!

File: backend/apps/owasp/api/search/chapter.py (hunk 2 above)
Comment (Rajgupta36) [reply backend/apps/owasp/api/search/chapter.py]: yes

**Review by Rajgupta36 — CHANGES_REQUESTED** [review]
Thanks 👍👍

**Review by arkid15r — APPROVED** [review]
I updated this a bit and merging into main
Thanks for another contribution @harsh3dev !

**Pull request merged**